numpy/fft/tests/test_pocketfft.py

```diff
@@ -200,6 +200,26 @@ def test_axes(self, op):
             tr_op = np.transpose(op(x, axes=a), a)
             assert_allclose(op_tr, tr_op, atol=1e-6)
 
+    @pytest.mark.parametrize("op", [np.fft.fftn, np.fft.ifftn])
+    def test_s_negative_1(self, op):
+        x = np.arange(100).reshape(10, 10)
+        # should use the whole input array along the first axis
+        assert op(x, s=(-1, 5), axes=(0, 1)).shape == (10, 5)
+
+    @pytest.mark.parametrize("op", [np.fft.fftn, np.fft.ifftn,
+                                    np.fft.rfftn, np.fft.irfftn])
+    def test_s_axes_none(self, op):
+        x = np.arange(100).reshape(10, 10)
+        with pytest.warns(match='`axes` should not be `None` if `s`'):
```

Comment:
Can use `pytest.deprecated_call` instead if that's preferred.

Reply:
We seem to use `pytest.warns` elsewhere, let's keep it consistent with the rest of the codebase.